include/libtorrent/hasher512.hpp

```diff
@@ -48,7 +48,14 @@ POSSIBILITY OF SUCH DAMAGE.
 
 #elif TORRENT_USE_CRYPTOAPI
 #include <windows.h>
-#include <wincrypt.h>
+#include <wincrypt.h> // CALG_SHA_512 conditionally defined here
+
+#ifdef CALG_SHA_512
+#define TORRENT_USE_CRYPTOAPI_SHA_512 1
```

Comment:
config.hpp used to include ``windows.h``, ideally it wouldn't though, to cut down on include cost/compile time. Couldn't you make this configuration depend on the windows version the build is targeting? i.e. ``_WIN32_WINNT``.